docs/NEW_NEAR_CLI_INTERFACE.md

```diff
@@ -0,0 +1,198 @@
+- `Tech-savvy` people automating their routines.
+- `Validators`
+
+`NEAR CLI UX principles`
+- All altering actions should have a confirmation step with an option to skip confirmation with an explicit command line parameter (e.g. `send` at the end of the command)
+- All direct children commands of a single parent command should be aligned (either represent an action or a resource, but never a mix of those on the same hierarchy level): `contract` -> `state` (resource) and `contract` -> `deploy` (action) are not aligned, so it should be either `contract` -> `get-state` + `contract` -> `deploy` or `contract` -> `state` -> `view` + `contract` -> `code` -> `deploy`
+- Interactive mode should look like: `command - description`. It will help people to learn the commands.
+
+## `Core NEAR CLI` commands
+
+```
+account
+  - view-account-summary <account-id> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+ 
+  - create-subaccount <new-account-id> <initial-balance> 
+    (we will treat everything after the first dot in account id as the parent account (transaction signer))
+    - autogenerate-new-keypair
+      - save-to-keychain network <"mainnet"|"testnet"|...>
+        - transaction signature options here (see below)
+      - print-to-terminal network <"mainnet"|"testnet"|...>
+        - transaction signature options here (see below)
+    - use-manually-provided-seed-prase "twelve words goes here" network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+    - use-manually-provided-public-key "ed25519:..." network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+
+  - delete-account <account-id> beneficiary <beneficiary-account-id> network <"mainnet"|"testnet"|...>
+    - transaction signature options here (see below)
+
+  - list-keys <account-id> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
+
+  - add-key <account-id>
+    - autogenerate-new-keypair network <"mainnet"|"testnet"|...>
+      - save-to-keychain network <"mainnet"|"testnet"|...>
+        - transaction signature options here (see below)
+      - print-to-terminal network <"mainnet"|"testnet"|...>
+        - transaction signature options here (see below)
+    - use-manually-provided-seed-prase "twelve words goes here" network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+    - use-manually-provided-public-key "ed25519:..." network <"mainnet"|"testnet"|...>
+      - transaction signature options here (see below)
+
+  - delete-key <account-id> <public-key> network <"mainnet"|"testnet"|...>
+    - transaction signature options here (see below)
+
+  - TODO: "import account" (aka "login") commands
+```
+
+```
+contract
+  - call-function
+    - as-read-only <account-id> <function-name> <function-args> network <"mainnet"|"testnet"|...> <now|at-timestamp|at-block-height|at-block-hash>
```

Comment:
Why not just `view`?  Even knowing that as-read-only means `view` I found it confusing and is less intuitive since we use `view` everywhere else.

```
near-cli contract call-function as-read-only ...
```

Is also a bit long.  I know you're love for being explicit, but having some aliases here would be nice.

```
near contract view
``` 

As a short cut would be very handy.

Reply:
To me just saying `view` might lead some people to think that they're viewing the contract code rather than it being viewing the result of a read-only function.

Reply:
If you execute a view call and try to invoke a host function like creating a transaction, you will get an error that says it was banned in a view call.  So my point is that we use `view` in our documentation and baked into the runtime, so using it, especially in this context helps people to learn it.

Reply:
@willemneal If you finish up the whole command, you will see that it does not make any sense to save a few characters:

```
near-cli contract view zavodil.poolv1.near get_owner_id '{}' network mainnet now
near-cli contract call-function as-read-only zavodil.poolv1.near get_owner_id '{}' network mainnet now
```

I don't expect anyone to type it out manually as we have interactive mode that guides users through it, and prints out the final command if you want to re-use it later.

`contract view` makes sense only to victims of near-cli-js naming. Anyone else will be confused and guess that this command will show you the contract, not calling a function. Yet, the good part of this new approach is that you will be able to implement your own extension and call it as `near-cli view ...` just fine, if you feel you want to have a shortcut.

Reply:
Hi, very nice PR. But i have a question.
Why not to use  `view` and `call` as argument for contract calling, I think not all people understand how read-only call works and `call-function as-read-only` may misunderstood people